api/document/tree.py

```diff
@@ -1,11 +1,13 @@
 from collections import Counter
-from typing import Callable, Optional
+from typing import Any, Callable, Dict, List, Optional
 
 from networkx import DiGraph
 from networkx.algorithms.dag import descendants
 
 from document.models import DocNode
 
+JsonDict = Dict[str, Any]
```

Comment:
We should probably swap `Json` for something else everywhere, lest we mislead. The deserialization process takes _in_ json, xml, etc. and spits out Python primitives. Would it make sense to use [`ReturnDict`](https://github.com/encode/django-rest-framework/blob/91fa8b923a51e51bafebcfa15e6e8ebab928642e/rest_framework/utils/serializer_helpers.py#L12) as our type?

Reply:
Oh funky.  Umm.  The main concept I wanted to convey was that the keys are `str` and the values can be ... well, I guess they can be any kind of data type that is serializable into a format that's like JSON, but might not necessarily be JSON?  Humm.

Reply:
Spit balling:
* Attrs
* StringMap
* Primitives
* POPOs
* ObjectData
* Parsed

Reply:
How about `PrimitiveDict`?  It's a bit wordy, but it seems kinda accurate...